client/packages/system/src/Item/Components/ItemVariant/PackUnitEntryCell.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { ReactElement, useState } from 'react';
+import {
+  RecordWithId,
+  CellProps,
+  Select,
+  PositiveNumberInput,
+  useDebounceCallback,
+  Box,
+} from '@openmsupply-client/common';
+import { useUnitVariant } from '../../context';
+
+// This cell displays a packSize number input and unit pack drop down if unit pack variants exist
+export const getPackUnitEntryCell =
+  <T extends RecordWithId>({
+    getItemId,
+    getUnitName,
+  }: {
+    getItemId: (row: T) => string;
+    getUnitName: (row: T) => string | null;
+  }) =>
+  ({ rowData, column }: CellProps<T>): ReactElement => {
+    const { asPackUnit, variantsControl } = useUnitVariant(
+      getItemId(rowData),
+      getUnitName(rowData)
+    );
+
+    const [packSize, setPackSize] = useState(
```

Comment:
I was wondering about useBufferState input, which looks like it's used for table cells. The comment in useBufferState suggest that we should avoid using it, and since cells won't be created unless rows exist, the initial value of the cell should always be present. Does that mean we shoulnd't use useBufferState for cells ?

Reply:
ideally shouldn't use it.. the warning from Josh is pretty clear. I can't see why it would be needed here. Seeding the state should be enough?

Reply:
> Seeding the state should be enough? 

Yeah i thought it would, unless the table is updated behind the scenes somehow (which I think might be bad UI/UX if it happens without warning or some conflict resolution)